### ISSUE TITLE
How to access multiple clusters

### ISSUE THREAD

**Issue by NasAmin** [kubernetes-client/csharp#446]
Hi,

I am trying to use this client to connect to multiple EKS clusters and then compare deployments across e.g. Dev/Staging/Prod clusters. 

I would like to deploy my dashboard to a kubernetes cluster. This dashboard needs to display workloads from all the "configured" clusters.

My plan was that I'll mount the kubeconfig for each cluster as a ConfigMap to the dashboard pod.
But I am struggling to get authentication working. 
My Kubeconfig uses aws cli for authentication. 
Is there a better way to access multiple clusters and get authenticated using the csharp client in a pod?

Regards,

Nas

**Comment by brendandburns** [comment]
Currently there is a 1-1 between auth and a client object, so you will need to create a client object for each Kubernetes cluster that you want to communicate with.

If you want to communicate with the cluster that the pod is running in, you should use Kubernetes service accounts, not the EKS/AWS credentials.

If you want to communicate with other clusters, you'll need to use the EKS/AWS credentials.

**Comment by NasAmin** [comment]
@brendanburns Thanks for the response.
I am not sure how to get the authentication working without kubectl proxy. I am getting this error:

```
k8s.Exceptions.KubeConfigException: 'User: arn:aws:eks:eu-west-2:<account name>:cluster/my-cluster does not have appropriate auth credentials in kubeconfig'

```

This is what the authentication in kubeconfig looks like
```
- name: arn:aws:eks:eu-west-2:<account name>:cluster/my-cluster
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1alpha1
      args:
      - --region
      - eu-west-2
      - eks
      - get-token
      - --cluster-name
      - my-cluster
      command: aws
      env: null
```

What auth method can I use that will work for external clusters without kubectl proxy?

Regards,

Nas

**Comment by brendandburns** [comment]
The relevent code is here:

**Comment by brendandburns** [comment]
See #450 I think that this is broken in dotnet 3.0 we'll see about removing the ifdef.

**Comment by KyleJFischer** [comment]
Hey just noticed this issue on here or else I would have commented, this is the same exact problem we were having. We were trying to connect to EKS and due to our application being .netcore 3 it wasn't working until I removed the ifdefs.

**Comment by NasAmin** [comment]
@brendanburns @KyleJFischer Thanks. Yes it does work with .netstandard2.0.
My app is .NET Core 3.1 that is why it didn't work.

It would be good if we can update to the .NET 3.1 given that .net Core 2.0/2.1 is deprecated!

Regards,

Nas

**Comment by NasAmin** [comment]
Closing this issue. The main issue about .NET Core 3.1 support was fixed on #450
